R/gh.R

```diff
@@ -195,11 +196,13 @@ gh <- function(endpoint, ..., per_page = NULL, .token = NULL, .destfile = NULL,
       res3 <- c(res, res2)      # e.g. GET /orgs/:org/invitations
     }
 
+    len <- len + gh_response_length(res2)
+
     attributes(res3) <- attributes(res2)
     res <- res3
   }
 
-  if (! is.null(.limit) && length(res) > .limit) {
+  if (! is.null(.limit) && len > .limit) {
     res_attr <- attributes(res)
     res <- res[seq_len(.limit)]
```

Comment:
There needs to be alternative behavior here since for the object case we need to subset the array within the object

Reply:
Yeah, but it is not entirely clear how to do that. This branch is only used if we got more items than requested, which might not even happen. So this is not super important I think. I'll protect this with `length(res) == len`, so we only do it in the non-named response case.

Reply:
I was thinking of a case where someone set `.limit = 101` in which case they would now get 200 results (not the end of the world).

Can we not use the same logic to find the "target" as the unnamed list and subset it?

Reply:
Yeah, OTOH subsetting based on some target list is potentially destructive, while just looking up the length os probably not. 

So I think we could only subset in the non-named case, to keep the old behavior for that. The behavior for the named case is much improved already, so it is fine I think.

Reply:
Sounds good - I think the usage of `.limit` is quite rare already and this should cover the vast majority of use cases